src/osd/OSD.cc

```diff
@@ -2130,6 +2130,10 @@ int OSD::write_meta(CephContext *cct, ObjectStore *store, uuid_d& cluster_fsid,
       return r;
   }
 
+  r = store->write_meta("created_ceph_version", pretty_version_to_str());
```

Comment:
Have you intentionally omitted this field retrieval in _collect_metadata? If so - why?

Reply:
ah... seen the comment above... IMO it makes sense to have this field in "ceph osd metadata" output as well.

Reply:
but may be rename to "ceph_version_when_created" ?

Reply:
Renamed and added to `_collect_metadata`. 

Should output:
```
# cat /var/lib/ceph/osd/ceph-32/{ceph_version_when_created,created_at}
ceph version 16.2.7-1084-g3b0c2317b26 (3b0c2317b265e917b7a79d87f5b5a9a8c770bbab) pacific (stable)
2022-10-05T15:18:42.962271Z

# ceph osd metadata 32| jq -r '[ .created_at, .ceph_version_when_created]'
[
  "2022-10-05T15:18:42.962271Z",
  "ceph version 16.2.7-1084-g3b0c2317b26 (3b0c2317b265e917b7a79d87f5b5a9a8c770bbab) pacific (stable)"
]
```